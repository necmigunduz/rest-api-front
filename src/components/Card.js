import PropTypes from 'prop-types';
import { useState } from 'react';

const Card = (props) => {
  const [value, setValue] = useState(0);
  const {
    id, title, submitHandler,
  } = props;

  const handleSubmit = (e, id, value) => {
    submitHandler(e.target.previousSibling.id, id, value);
  };

  const handleChange = (e) => {
    setValue(e.target.value);
  };

  return (
    <div className="add-card flex space-between align-center m-b-20 p-l-10 p-r-10">
      <div className="color-gray medium input-title flex">{title}</div>
      <input className="input" id={title} type="number" placeholder="0" value={value} onChange={(e) => handleChange(e)} />
      <button className="submit-button color-dark-gray medium" type="submit" onClick={(e) => handleSubmit(e, id, value)}>+</button>
    </div>
  );
};

Card.propTypes = {
  id: PropTypes.number.isRequired,
  title: PropTypes.string.isRequired,
  submitHandler: PropTypes.func.isRequired,
};

export default Card;
