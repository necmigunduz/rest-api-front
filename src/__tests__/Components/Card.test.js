import renderer from 'react-test-renderer';
import Card from '../../components/Card';

describe('Card', () => {
  it('renders correctly', () => {
    const tree = renderer.create(
      <Card
        id={2}
        title="CardTitle"
        value="50"
        submitHandler={() => {}}
        changeHandler={() => {}}
      />,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
