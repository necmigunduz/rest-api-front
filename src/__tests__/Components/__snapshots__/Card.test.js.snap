// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card renders correctly 1`] = `
<div
  className="add-card flex space-between align-center m-b-20 p-l-10 p-r-10"
>
  <div
    className="color-gray medium input-title flex"
  >
    CardTitle
  </div>
  <input
    className="input"
    id="CardTitle"
    onChange={[Function]}
    placeholder="0"
    type="number"
    value="50"
  />
  <button
    className="submit-button color-dark-gray medium"
    onClick={[Function]}
    type="submit"
  >
    +
  </button>
</div>
`;
